### PR TITLE
Update README.md

### LINKED ISSUE #10: Is the PermonFLLOP link in README.md still relevant?
Maybe it should be mentioned as just a part of PermonQP.

### PR DESCRIPTION
**Pull request by haplav**
Fix #10.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,13 @@
 PERMON toolbox
 ================================
-PERMON (Parallel Efficient Robust Modular Object Numerical) is a software toolbox for quadratic programming (QP) based on [PETSc](http://www.mcs.anl.gov/petsc/). It also includes domain decomposition methods (FETI and Total FETI) as a specific QP implementation.
+PERMON (Parallel Efficient Robust Modular Object Numerical) is a software toolbox for quadratic programming (QP) based on [PETSc](http://www.mcs.anl.gov/petsc/). It also includes domain decomposition methods (FETI and Total FETI) and Support Vector Machines (SVM) machine learning methods.
```
Comment (jkruzik) [inline README.md]: It should be clear that this repository only contains QP stuff and FETI

File: README.md [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
 
 PERMON contains following modules:
 
-* [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP transformations and solvers
-* [PermonFLLOP](http://permon.vsb.cz/permonfllop.htm) - (FETI Light Layer On top of PETSc) implements (Total) FETI
-
-Support Vector Machines (SVMs) implementation [PermonSVM](http://permon.vsb.cz/permonsvm.htm) based on PermonQP can be found at <https://github.com/permon/permonsvm>.
-
+* [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP solvers and (Total) FETI domain decomposition methods
+* [PermonSVM](http://permon.vsb.cz/permonsvm.htm) - SVM machine learning implementation based on PermonQP.
```
Comment (jkruzik) [inline README.md]: There is no link to PermonSVM respository

File: README.md [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
 Project homepage: <http://permon.vsb.cz>  
 
 PERMON contains following modules:
 
-* [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP transformations and solvers
-* [PermonFLLOP](http://permon.vsb.cz/permonfllop.htm) - (FETI Light Layer On top of PETSc) implements (Total) FETI
-
-Support Vector Machines (SVMs) implementation [PermonSVM](http://permon.vsb.cz/permonsvm.htm) based on PermonQP can be found at <https://github.com/permon/permonsvm>.
-
+* [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP solvers and (Total) FETI domain decomposition methods
```
Comment (haplav) [inline README.md]:
```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) (this repository) - contains QP solvers and (Total) FETI domain decomposition methods
```

File: README.md (hunk 2 above)
Comment (haplav) [reply README.md]:
```suggestion
* [PermonSVM](http://permon.vsb.cz/permonsvm.htm) [(another repo)](https://github.com/permon/permonsvm) - SVM machine learning implementation based on PermonQP.
```

File: README.md (hunk 1 above)
Comment (haplav) [reply README.md]: Would this suffice? https://github.com/permon/permon/pull/14/files#r417501398

File: README.md (hunk 3 above)
Comment (jkruzik) [inline README.md]:
```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) - solvers and routines for QP problems and (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods
```

The fllop web page should be eventually changed to FETI which should be a subsection of PermonQP.

File: README.md (hunk 2 above)
Comment (jkruzik) [inline README.md]:
```suggestion
* [PermonSVM](http://permon.vsb.cz/permonsvm.htm) - SVM machine learning implementation based on PermonQP - [separate repository](https://github.com/permon/permonsvm)
```

File: README.md (hunk 1 above)
Comment (jkruzik) [reply README.md]: It is probably sufficient that it is in the repository description.

I suggest changing the repository description to  "PERMON main package for quadratic programming (PermonQP)".

Comment (jkruzik) [inline README.md]:
```suggestion
PERMON (Parallel Efficient Robust Modular Object Numerical) is a software toolbox for the scalable (parallel) solution of  quadratic programming (QP) problems based on [PETSc](http://www.mcs.anl.gov/petsc/). It also includes domain decomposition methods (FETI and Total FETI) and Support Vector Machines (SVM) machine learning methods.
```

Comment (jkruzik) [reply README.md]: > It also includes

It may seem a bit random that a QP toolbox includes these. Should we somehow clarify that these are QP problems?

Comment (jkruzik) [reply README.md]: > FETI

Shouldn't we stress that the combination with the QP solvers gives the ability to (efficiently) solve contact problems?

Comment (jkruzik) [reply README.md]: > > FETI
> 
> Shouldn't we stress that the combination with the QP solvers gives the ability to (efficiently) solve contact problems?

Also maybe FETI can be subpoint in the PermonQP point

File: README.md (hunk 3 above)
Comment (haplav) [reply README.md]: superseded by https://github.com/permon/permon/pull/14#discussion_r419310866

File: README.md [hunk 4]
```diff
@@ -1,16 +1,13 @@
...
 Project homepage: <http://permon.vsb.cz>  
 
 PERMON contains following modules:
 
-* [PermonQP](http://permon.vsb.cz/permonqp.htm) - contains QP transformations and solvers
-* [PermonFLLOP](http://permon.vsb.cz/permonfllop.htm) - (FETI Light Layer On top of PETSc) implements (Total) FETI
-
-Support Vector Machines (SVMs) implementation [PermonSVM](http://permon.vsb.cz/permonsvm.htm) based on PermonQP can be found at <https://github.com/permon/permonsvm>.
-
+* [PermonQP](http://permon.vsb.cz/permonqp.htm) - solvers and routines for QP problems and (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods
```
Comment (haplav) [inline README.md]:
```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) - this repository
  - solvers and routines for solution of QP problems
  - constrained (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods - solve contact problems with domain decomposition
```

Comment (haplav) [reply README.md]: @jkruzik

Comment (jkruzik) [reply README.md]: We could even drop "this repository".
I would suggest changing "domain decomposition methods - solve contact problems with domain decomposition" to
"domain decomposition methods able to solve contact problems"

Comment (haplav) [reply README.md]: > We could even drop "this repository".

I would like to keep it for clarity as long as the PermonQP name is used. But maybe we could abandon this name. It's just PERMON, and PermonSVM is its extension for SVM.

> I would suggest changing "domain decomposition methods - solve contact problems with domain decomposition" to
"domain decomposition methods able to solve contact problems"

Updated suggestion:
```suggestion
* [PermonQP](http://permon.vsb.cz/permonqp.htm) - this repository
  - solvers and routines for solution of QP problems
  - constrained (Total) [FETI](http://permon.vsb.cz/permonfllop.htm) domain decomposition methods able to solve contact problems
```

Comment (jkruzik) [reply README.md]: OK, keep it.
> It's just PERMON...

You can suggest it by mail to the others.

File: README.md [hunk 5]
```diff
@@ -35,4 +32,3 @@ The documentation of the routines is available at <http://permon.vsb.cz/document
 Supported PETSc versions
 ----------------------------------
 PERMON tries to support newest versions of PETSc as soon as possible. The [releases](https://github.com/permon/permon/releases) are tagged with major.minor.sub-minor numbers. The major.minor numbers correspond to the major.minor release numbers of the supported PETSc version.
-
```
Comment (jkruzik) [inline README.md:38]: There should be a new line here.

Comment (haplav) [reply README.md:38]: I wonder why but that's not worth further discussion.

Comment (haplav) [reply README.md:38]: If I try to edit the file, the newline is there. It seems it's just some GitHub noise. Closing.

Comment (jkruzik) [reply README.md:38]: I think that GitHub shows the new line as missing making the last line not being a POSIX line. Not a big deal in README. It should not happen in a code as some compilers still throw warnings (e.g., CRAY).

**Pull request merged**